### PR TITLE
depends|build: Make sure configuration detects Qt built with DEBUG=1

### PR DESCRIPTION
**Pull request by xdustinface**
Backporting some stuff ahead of time to fix Qt config if built with `DEBUG=1` which seems to be broken (at least for me here) since we upgraded to Qt 5.9.6. Note that bitcoin#18298 isn't merged yet to upstream but it seems to work fine. 

Tested on macOS 10.15.6, no idea what Windows thinks about this but at least CI still passes.

### CODE REVIEW

File: build-aux/m4/bitcoin_qt.m4 [hunk 1]
```diff
@@ -302,49 +307,49 @@ AC_DEFUN([_BITCOIN_QT_FIND_STATIC_PLUGINS],[
...
-        fi
-      ])
+      PKG_CHECK_MODULES([QTFONTDATABASE], [Qt5FontDatabaseSupport], [QT_LIBS="-lQt5FontDatabaseSupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTEVENTDISPATCHER], [Qt5EventDispatcherSupport], [QT_LIBS="-lQt5EventDispatcherSupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTTHEME], [Qt5ThemeSupport], [QT_LIBS="-lQt5ThemeSupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTDEVICEDISCOVERY], [Qt5DeviceDiscoverySupport], [QT_LIBS="-lQt5DeviceDiscoverySupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTACCESSIBILITY], [Qt5AccessibilitySupport], [QT_LIBS="-lQt5AccessibilitySupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTFB], [Qt5FbSupport], [QT_LIBS="-lQt5FbSupport $QT_LIBS"])
+      if test "x$TARGET_OS" = xlinux; then
+        PKG_CHECK_MODULES([QTXCBQPA], [Qt5XcbQpa], [QT_LIBS="$QTXCBQPA_LIBS $QT_LIBS"])
```
Comment (UdjinM6) [inline build-aux/m4/bitcoin_qt.m4:324]:
```suggestion
        PKG_CHECK_MODULES([X11XCB], [x11-xcb], [QT_LIBS="$X11XCB_LIBS $QT_LIBS"])
        PKG_CHECK_MODULES([QTXCBQPA], [Qt5XcbQpa], [QT_LIBS="$QTXCBQPA_LIBS $QT_LIBS"])
```

or backport https://github.com/bitcoin/bitcoin/pull/16408

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: build-aux/m4/bitcoin_qt.m4 (hunk 1 above)
Comment (xdustinface) [reply build-aux/m4/bitcoin_qt.m4:324]: Added bitcoin#15978 and bitcoin#16408

File: depends/packages/packages.mk [hunk 2]
```diff
@@ -3,7 +3,7 @@ packages:=boost openssl libevent zeromq gmp chia_bls backtrace
 qt_native_packages = native_protobuf
 qt_packages = qrencode protobuf zlib
 
-qt_linux_packages:=qt expat dbus libxcb xcb_proto libXau xproto freetype fontconfig libX11 xextproto libXext xtrans
+qt_linux_packages:=qt expat dbus libxcb xcb_proto libXau xproto freetype fontconfig
```
Comment (UdjinM6) [inline depends/packages/packages.mk:6]: should drop `depends/packages/libX11.mk`, `depends/packages/libXext.mk`, `depends/packages/xextproto.mk` and `depends/packages/xtrans.mk` _files_ too

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: depends/packages/packages.mk (hunk 2 above)
Comment (xdustinface) [reply depends/packages/packages.mk:6]: Good point 😄 Done with the last force push.

**Review by UdjinM6 — APPROVED** [review]
Works for me (but only when rebased on top of #4003)

ACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**